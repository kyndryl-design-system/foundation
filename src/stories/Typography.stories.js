import { html } from 'lit';
import { classMap } from 'lit/directives/class-map.js';

export default {
  title: 'Foundation/Typography',
};

export const Hero = {
  argTypes: {
    fontWeight: {
      options: ['ultralight', 'light', 'bold'],
      control: { type: 'select' },
      description: 'Change font weight.',
      table: {
        defaultValue: { summary: 'ultralight' },
      },
    },
  },
  args: {
    fontWeight: 'ultralight',
  },
  render: (args) => {
    const weightClasses = {
      'kd-type--weight-ultralight': args.fontWeight == 'ultralight',
      'kd-type--weight-light': args.fontWeight == 'light',
      'kd-type--weight-regular': args.fontWeight == 'regular',
      'kd-type--weight-medium': args.fontWeight == 'medium',
      'kd-type--weight-bold': args.fontWeight == 'bold',
    };

    return html`
      <div class="type-examples">
        <div class="kd-type--hero-01 ${classMap(weightClasses)}">Hero-01</div>
        <div class="kd-type--hero-02 ${classMap(weightClasses)}">Hero-02</div>
        <div class="kd-type--hero-03 ${classMap(weightClasses)}">Hero-03</div>
        <div class="kd-type--hero-04 ${classMap(weightClasses)}">Hero-04</div>
      </div>
    `;
  },
};

export const Display = {
  argTypes: {
    fontWeight: {
      options: ['ultralight', 'light'],
      control: { type: 'select' },
      description: 'Change font weight.',
      table: {
        defaultValue: { summary: 'ultralight' },
      },
    },
  },
  args: {
    fontWeight: 'ultralight',
  },
  render: (args) => {
    const weightClasses = {
      'kd-type--weight-ultralight': args.fontWeight == 'ultralight',
      'kd-type--weight-light': args.fontWeight == 'light',
    };

    return html`
      <div class="type-examples">
        <div class="kd-type--display-01 ${classMap(weightClasses)}">
          Display-01
        </div>
        <div class="kd-type--display-02 ${classMap(weightClasses)}">
          Display-02
        </div>
        <div class="kd-type--display-03 ${classMap(weightClasses)}">
          Display-03
        </div>
      </div>
    `;
  },
};

export const Headline = {
  argTypes: {
    fontWeight: {
      options: ['ultralight', 'light', 'regular'],
      control: { type: 'select' },
      description: 'Change font weight.',
      table: {
        defaultValue: { summary: '1-3=light, 4-8=regular' },
      },
    },
  },
  args: {
    fontWeight: 'light',
  },
  render: (args) => {
    const weightClasses = {
      'kd-type--weight-ultralight': args.fontWeight == 'ultralight',
      'kd-type--weight-light': args.fontWeight == 'light',
      'kd-type--weight-regular': args.fontWeight == 'regular',
    };

    return html`
      <div class="type-examples">
        <div class="kd-type--headline-01 ${classMap(weightClasses)}">
          Headline-01
        </div>
        <div class="kd-type--headline-02 ${classMap(weightClasses)}">
          Headline-02
        </div>
        <div class="kd-type--headline-03 ${classMap(weightClasses)}">
          Headline-03
        </div>
        <div class="kd-type--headline-04 ${classMap(weightClasses)}">
          Headline-04
        </div>
        <div class="kd-type--headline-05 ${classMap(weightClasses)}">
          Headline-05
        </div>
        <div class="kd-type--headline-06 ${classMap(weightClasses)}">
          Headline-06
        </div>
        <div class="kd-type--headline-07 ${classMap(weightClasses)}">
          Headline-07
        </div>
        <div class="kd-type--headline-08 ${classMap(weightClasses)}">
          Headline-08
        </div>
      </div>
    `;
  },
};

export const Body = {
  argTypes: {
    fontWeight: {
      options: ['light', 'regular', 'medium'],
      control: { type: 'select' },
      description: 'Change font weight.',
      table: {
        defaultValue: { summary: 'regular' },
      },
    },
  },
  args: {
    fontWeight: 'regular',
  },
  render: (args) => {
    const weightClasses = {
      'kd-type--weight-light': args.fontWeight == 'light',
      'kd-type--weight-regular': args.fontWeight == 'regular',
      'kd-type--weight-medium': args.fontWeight == 'medium',
    };

    return html`
      <div class="type-examples">
        <div class="kd-type--body-01 ${classMap(weightClasses)}">Body-01</div>
        <div class="kd-type--body-02 ${classMap(weightClasses)}">Body-02</div>
      </div>
    `;
  },
};

export const Utility = {
  argTypes: {
    fontWeight: {
      options: ['light', 'regular', 'medium'],
      control: { type: 'select' },
      description: 'Change font weight.',
      table: {
        defaultValue: { summary: 'regular' },
      },
    },
  },
  args: {
    fontWeight: 'regular',
  },
  render: (args) => {
    const weightClasses = {
      'kd-type--weight-light': args.fontWeight == 'light',
      'kd-type--weight-regular': args.fontWeight == 'regular',
      'kd-type--weight-medium': args.fontWeight == 'medium',
    };

    return html`
      <div class="type-examples">
        <div class="kd-type--ui-01 ${classMap(weightClasses)}">UI-01</div>
        <div class="kd-type--ui-02 ${classMap(weightClasses)}">UI-02</div>
        <div class="kd-type--ui-03 ${classMap(weightClasses)}">UI-03</div>
        <div class="kd-type--ui-04 ${classMap(weightClasses)}">UI-04</div>
      </div>
    `;
  },
};
