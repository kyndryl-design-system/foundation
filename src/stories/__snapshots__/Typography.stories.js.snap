// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Foundation/Typography Body smoke-test 1`] = `
<div class="type-examples">
  <div class="kd-type--body-01   ">
    Body-01
  </div>
  <div class="kd-type--body-02   ">
    Body-02
  </div>
</div>
`;

exports[`Foundation/Typography Display smoke-test 1`] = `
<div class="type-examples">
  <div class="kd-type--display-01   ">
    Display-01
  </div>
  <div class="kd-type--display-02   ">
    Display-02
  </div>
  <div class="kd-type--display-03   ">
    Display-03
  </div>
</div>
`;

exports[`Foundation/Typography Headline smoke-test 1`] = `
<div class="type-examples">
  <div class="kd-type--headline-01   ">
    Headline-01
  </div>
  <div class="kd-type--headline-02   ">
    Headline-02
  </div>
  <div class="kd-type--headline-03   ">
    Headline-03
  </div>
  <div class="kd-type--headline-04   ">
    Headline-04
  </div>
  <div class="kd-type--headline-05   ">
    Headline-05
  </div>
  <div class="kd-type--headline-06   ">
    Headline-06
  </div>
  <div class="kd-type--headline-07   ">
    Headline-07
  </div>
  <div class="kd-type--headline-08   ">
    Headline-08
  </div>
</div>
`;

exports[`Foundation/Typography Hero smoke-test 1`] = `
<div class="type-examples">
  <div class="kd-type--hero-01   ">
    Hero-01
  </div>
  <div class="kd-type--hero-02   ">
    Hero-02
  </div>
  <div class="kd-type--hero-03   ">
    Hero-03
  </div>
  <div class="kd-type--hero-04   ">
    Hero-04
  </div>
</div>
`;

exports[`Foundation/Typography Utility smoke-test 1`] = `
<div class="type-examples">
  <div class="kd-type--ui-01   ">
    UI-01
  </div>
  <div class="kd-type--ui-02   ">
    UI-02
  </div>
  <div class="kd-type--ui-03   ">
    UI-03
  </div>
  <div class="kd-type--ui-04   ">
    UI-04
  </div>
  <div class="kd-type--ui-05   ">
    UI-05
  </div>
</div>
`;
